src/DataLayouts/DataLayouts.jl

```diff
@@ -92,6 +92,21 @@ abstract type AbstractDispatchToDevice end
 struct ToCPU <: AbstractDispatchToDevice end
 struct ToCUDA <: AbstractDispatchToDevice end
 
+"""
+    AbstractMask
+
+An abstract mask type, for marking domains as present or absent.
+"""
+abstract type AbstractMask end
```

Comment:
We should add documentation of properties/methods a type that subtypes `AbstractMask` should implement.

Reply:
Yes, I'm going to start adding more docs now, since some things have changed since the initial sketch (in order to avoid compute on columns in the cuda kernels).

Reply:
I added some docs that refer to this in the developer docs, but this is really an internal type, since the only two things that users need are: `enable_mask = true` in the space constructor, and `set_mask!`